src/libraries/component-time-scroll-view/TimeScrollViewDimensions.ts

```diff
@@ -75,9 +75,13 @@ export const usePanelDimensions = (width: number, height: number, panelCount: nu
 
 export const useFocusTimeInPixels = (timeToPixelMatrix: Matrix) => {
     const {focusTime} = useTimeFocus()
+    const {visibleTimeStartSeconds, visibleTimeEndSeconds} = useTimeRange()
     const pixelTime = useMemo(() => {
         if (focusTime === undefined) return undefined
+        if ((focusTime < (visibleTimeStartSeconds || 0)) || (focusTime > (visibleTimeEndSeconds || 0))) {
+            return undefined
+        }
         return convert1dDataSeries([focusTime], timeToPixelMatrix)[0]
-    }, [timeToPixelMatrix, focusTime])
+    }, [timeToPixelMatrix, focusTime, visibleTimeStartSeconds, visibleTimeEndSeconds])
```

Comment:
I think rather than increasing the number of dependencies here, I'd be more inclined to add a `timeFocusIsVisible` property to the return of the `useTimeFocus` hook. I'll put in a PR for that.